### PR TITLE
Fixed issue #18995: Confidentiality issue about username

### PR DESCRIPTION
**Pull request by gabrieljenik**

### CODE REVIEW

File: application/models/User.php [hunk 1]
```diff
@@ -981,4 +981,48 @@ public function canEdit($loginId)
...
+        return $criteriaPerm;
+    }
+
+    /**
+     * Uses for scope in list and search
+     * This don't mean user are allowed to read all information or any other Permission
+     * @param integer $userid
+     * @return self
+     */
+    public function withListRight($userid = null)
```
Comment (Shnoulle) [inline application/models/User.php:1023]: Oh ?

**Review by Shnoulle — APPROVED** [review]

**Review by Shnoulle — CHANGES_REQUESTED** [review]
Need to be in model too : https://github.com/LimeSurvey/LimeSurvey/blob/5c4fb4e0137d8e564c14d30c3c9563ffd7ecaacd/application/helpers/remotecontrol/remotecontrol_handle.php#L409C31-L409C31

**Comment by tiborpacalat** [comment]
And some checks are failing. Please make it green :)

**Comment by gabrieljenik** [comment]
> Need to be in model too : https://github.com/LimeSurvey/LimeSurvey/blob/5c4fb4e0137d8e564c14d30c3c9563ffd7ecaacd/application/helpers/remotecontrol/remotecontrol_handle.php#L409C31-L409C31

What do you mean in model? Check ownership permission in model?
I think usually those checks are outside of the model, right?
I would just add the check on the RC if that's what you mean

**Comment by Shnoulle** [comment]
> > Need to be in model too : https://github.com/LimeSurvey/LimeSurvey/blob/5c4fb4e0137d8e564c14d30c3c9563ffd7ecaacd/application/helpers/remotecontrol/remotecontrol_handle.php#L409C31-L409C31
> 
> What do you mean in model? Check ownership permission in model? I think usually those checks are outside of the model, right? I would just add the check on the RC if that's what you mean

It's a question, not an affirmation. ping @olleharstedt 

In model : it fix for all controller.

Else here : must be reported for RC too.

**Review by Shnoulle — APPROVED** [review]

**Pull request merged**